transmit-lib/src/cmr/transmit/config.clj

```diff
@@ -244,10 +248,20 @@
   {:connection-manager (conn/conn-mgr connection)
    :socket-timeout (http-socket-timeout)})
 
-(defn application-public-root-url
-  "Returns the public root url for an application given a context. Assumes public configuration is
-   stored in a :public-conf key of the system."
-  [context]
-  (let [{:keys [protocol host port relative-root-url]} (get-in context [:system :public-conf])
-        port (if (empty? relative-root-url) port (format "%s%s" port relative-root-url))]
+(defn- format-public-root-url
+  [{:keys [protocol host port relative-root-url]}]
+  (let [port (if (empty? relative-root-url) port (format "%s%s" port relative-root-url))]
```

Comment:
nit: this is confusing, I know it is existing code.

Why not get rid of this incorrectly named `port` binding and change the line below to:
```(format "%s://%s:%s%s" protocol host port (or relative-root-url "")))```
or bind relative-root-url to `(or relative-root-url "")`

Reply:
Yeah, I've never like this code. It's awkward mostly because network location URLs are awkward and chock 'o block full of special cases. I think when this was written the focus was on least amount of code/bindings possible. I'll take a look.

Reply:
Okay, added some tests and then opted for a default value of `relative-root-url` in the args and then a simple and easy-to-read conditional in the body. It's a little more verbose, but you know exactly what it's doing.

Reply:
All tests pass locally -- pushing ...